src/components/Task/Modal/Form/TaskModalHelper.ts

```diff
@@ -115,13 +117,19 @@ export const handleResultChange = ({
     result as DisplayResultEnum,
     ActivityTypeEnum.None,
   );
+  const defaultNextAction = findNextAction(completedAction, nextActions);
+  setFieldValue('nextAction', defaultNextAction);
+};
+
+const findNextAction = (completedAction, nextActions) => {
   const actionsWithoutNone = nextActions.filter(
     (action) => action !== ActivityTypeEnum.None,
   );
-  setFieldValue(
-    'nextAction',
-    actionsWithoutNone.length === 1 ? actionsWithoutNone[0] : undefined,
-  );
+  return completedAction && nextActions.includes(completedAction)
```

Comment:
I feel like this could be easier to read with if/else if/else instead of a ternary.